vars/cloudFoundryDeploy.groovy

```diff
@@ -157,54 +187,70 @@ void call(Map parameters = [:]) {
-                    .use()
-
-                dockerExecute(script: script, dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent) {
-                    deployMta(config)
-                }
+                deployTriggered = true
+                handleMTADeployment(config, script)
             }
-
-            if (config.deployTool == 'cf_native') {
-                deploy = true
-                config.smokeTest = ''
-
-                if (config.smokeTestScript == 'blueGreenCheckScript.sh') {
-                    writeFile file: config.smokeTestScript, text: libraryResource(config.smokeTestScript)
-                }
-
-                config.smokeTest = '--smoke-test $(pwd)/' + config.smokeTestScript
-                sh "chmod +x ${config.smokeTestScript}"
-
-                echo "[${STEP_NAME}] CF native deployment (${config.deployType}) with cfAppName=${config.cloudFoundry.appName}, cfManifest=${config.cloudFoundry.manifest}, smokeTestScript=${config.smokeTestScript}"
-
-                dockerExecute (
-                    script: script,
-                    dockerImage: config.dockerImage,
-                    dockerWorkspace: config.dockerWorkspace,
-                    stashContent: config.stashContent,
-                    dockerEnvVars: [CF_HOME:"${config.dockerWorkspace}", CF_PLUGIN_HOME:"${config.dockerWorkspace}", STATUS_CODE: "${config.smokeTestStatusCode}"]
-                ) {
-                    deployCfNative(config)
-                }
+            else if (config.deployTool == 'cf_native') {
+                deployTriggered = true
+                handleCFNativeDeployment(config, script)
+            }
+            else {
+                deployTriggered = false
+                echo "[${STEP_NAME}] WARNING! Found unsupported deployTool. Skipping deployment."
             }
         } catch (err) {
             deploySuccess = false
             throw err
         } finally {
-            if (deploy) {
+            if (deployTriggered) {
                 reportToInflux(script, config, deploySuccess, jenkinsUtils)
             }
 
         }
+    }
+}
+
+private void handleMTADeployment(Map config, script) {
+    // set default mtar path
+    config = ConfigurationHelper.newInstance(this, config)
+        .addIfEmpty('mtaPath', config.mtaPath ?: findMtar())
+        .use()
+
+    dockerExecute(script: script, dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent) {
+        deployMta(config)
+    }
+}
+
+private void handleCFNativeDeployment(Map config, script) {
+    config.smokeTest = ''
+
+    if (config.deployType == 'blue-green') {
+        prepareBlueGreenCfNativeDeploy(config,script)
+    } else {
+        config.deployCommand = 'push'
+        config.deployOptions = "${varOptions(config)}${varFileOptions(config)}"
+    }
 
+    echo "[${STEP_NAME}] CF native deployment (${config.deployType}) with:"
```

Comment:
could be put into multi-line string with one `echo`
Otherwise it appears a bit cluttered in the log